rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,345 @@
+from icon_rapid7_insight_agent.util.graphql_api.region_map import region_map
+from icon_rapid7_insight_agent.util.graphql_api.api_exception import APIException
+from typing import Any
+import icon_rapid7_insight_agent.util.graphql_api.agent_typer as agent_typer
+import requests
+import logging
+
+
+class ApiConnection:
+    """
+    ApiConnection(api_key, region_string, logger)
+
+    A class to connect to the Insight Agent GraphQL. This class provides convince methods to perform actions
+    on Insight Agents.
+    """
+    def __init__(self,
+                 api_key: str,
+                 region_string: str,
+                 logger: logging.log
+                 ) -> None:
+
+        self.api_key = api_key
+        self.logger = logger
+        self.endpoint = self._setup_endpoint(region_string)
+
+        self.session = requests.Session()
+        self.session.headers = self._get_headers()
+
+        self.org_key = self._get_org_key()
+        self.logger.info(f"Received org key: ********-****-****-****-*******{self.org_key[-5:]}")
```

Comment:
This is better suited for a classmethod - possibly "set up" the variables that can't be passed into the __init__ (ie. require function calls) and then pass the final variables into the init, but do the logging from the classmethod